File: app/src/main/java/com/ladsoft/bakingapp/data/database/translator/StepRecordTranslator.java
package com.ladsoft.bakingapp.data.database.translator;

import com.ladsoft.bakingapp.data.database.entity.StepRecord;
import com.ladsoft.bakingapp.entity.Step;
import com.ladsoft.bakingapp.translator.Translator;

public class StepRecordTranslator implements Translator<Step, StepRecord> {
    @Override
    public StepRecord translate(Step source) {
        return new StepRecord(source.getId(),
                        source.getRecipeId(),
                        source.getShortDescription() != null ? source.getShortDescription() : "",
                        source.getDescription() != null ? source.getDescription() : "",
                        source.getVideoUrl() != null ? source.getVideoUrl() : "",
                        source.getThumbnailUrl() != null ? source.getThumbnailUrl() : ""
                    );
    }
}

File: app/src/main/java/com/ladsoft/bakingapp/data/database/translator/StepRecordTranslator.kt
package com.ladsoft.bakingapp.data.database.translator

import com.ladsoft.bakingapp.data.database.entity.StepRecord
import com.ladsoft.bakingapp.entity.Step
import com.ladsoft.bakingapp.translator.Translator

class StepRecordTranslator: Translator<Step, StepRecord> {
    override fun translate(source: Step): StepRecord{
        return StepRecord(source.id, source.recipeId, source.stepId, source.shortDescription, source.description, source.videoUrl, source.thumbnailUrl)
    }
}